shell/platform/android/io/flutter/plugin/text/ProcessTextPlugin.java

```diff
@@ -0,0 +1,192 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugin.text;
+
+import android.annotation.TargetApi;
+import android.app.Activity;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.content.pm.ResolveInfo;
+import android.os.Build;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.RequiresApi;
+import io.flutter.embedding.engine.plugins.FlutterPlugin;
+import io.flutter.embedding.engine.plugins.activity.ActivityAware;
+import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding;
+import io.flutter.embedding.engine.systemchannels.ProcessTextChannel;
+import io.flutter.plugin.common.MethodChannel;
+import io.flutter.plugin.common.PluginRegistry.ActivityResultListener;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ProcessTextPlugin
+    implements FlutterPlugin,
+        ActivityAware,
+        ActivityResultListener,
+        ProcessTextChannel.ProcessTextMethodHandler {
+  private static final String TAG = "ProcessTextPlugin";
+
+  @NonNull private final ProcessTextChannel processTextChannel;
+  @NonNull private final PackageManager packageManager;
+  @Nullable private ActivityPluginBinding activityBinding;
+  private Map<String, ResolveInfo> resolveInfosById;
+
+  @NonNull
+  private Map<Integer, MethodChannel.Result> requestsByCode =
+      new HashMap<Integer, MethodChannel.Result>();
+
+  public ProcessTextPlugin(@NonNull ProcessTextChannel processTextChannel) {
+    this.processTextChannel = processTextChannel;
+    this.packageManager = processTextChannel.packageManager;
+
+    processTextChannel.setMethodHandler(this);
+  }
+
+  @Override
+  public Map<String, String> queryTextActions() {
+    if (resolveInfosById == null) {
+      resolveInfosById = new HashMap<String, ResolveInfo>();
```

Comment:
Nit: can this happen in `cacheResolveInfos` versus here?

Reply:
Yes, great suggestion, and doing so we can remove the `resolveInfosById.clear()`call your mention below.